### PR TITLE
Retry syncing geth if crashes in e2e test

### PR DESCRIPTION
**Pull request by jfoutts-celo**
### Description

Adds `initAndSyncGethWithRetry` to replace the sequence of `initAndStartGeth`, `connectPeers` and `waitToFinishInstanceSyncing`. 

### Tested

Tested celotool Docker still works & no failures in 200 runs:
```
for ((i=0; i<200; i++)); do ./ci_test_sync.sh local ~/celo-blockchain/ | tee run$i.txt ; done
```

### Other changes

Address other indeterminacy in sync_tests by removing `sleep`s.

### Related issues

Part of Epoch #847: Builds reliably green

### Backwards compatibility

Yes

### CODE REVIEW

File: packages/celotool/src/e2e-tests/utils.ts [hunk 1]
```diff
@@ -32,6 +35,37 @@ import { ensure0x, spawnCmd, spawnCmdWithExitOnFailure } from '../lib/utils'
...
+    try {
+      await initAndStartGeth(gethConfig, gethBinaryPath, instance, verbose)
+      await connectPeers(connectInstances, verbose)
+      await waitToFinishInstanceSyncing(instance)
+      break
+    } catch (error) {
+      console.info(`initAndSyncGethWithRetry error: ${error}`)
+      const logFilename = getLogFilename(gethConfig.runPath, instance)
+      console.info(`tail -50 ${logFilename}`)
+      console.info(await readLastLines.read(logFilename, 50))
```
Comment (mcortesi) [inline packages/celotool/src/e2e-tests/utils.ts:56]: do we have any idea of why we need to retry?

File: packages/celotool/src/types.d.ts [hunk 2]
```diff
@@ -1,2 +1,8 @@
 declare module 'web3-utils'
 declare module 'country-data'
+declare module 'read-last-lines' {
```
Comment (mcortesi) [inline packages/celotool/src/types.d.ts:3]: there isn't a `@types/read-last-lines`?

**Review by mcortesi — APPROVED** [review]
Amazing, thanks for working on this

File: packages/celotool/src/types.d.ts (hunk 2 above)
Comment (jfoutts-celo) [reply packages/celotool/src/types.d.ts:3]: No, unfortunately.  And then the TS_NODE_FILES=true is needed to pickup this type.

File: packages/celotool/src/e2e-tests/utils.ts (hunk 1 above)
Comment (jfoutts-celo) [reply packages/celotool/src/e2e-tests/utils.ts:56]: It seems to have been modulated by the `killGeth()` changes, and now happens less frequently.  Here's example (they all seem to look the same):
```
initAndSyncGethWithRetry error: Error: Invalid JSON RPC response: ""
tail -50 /tmp/e2e/fast/datadir/logs.txt
ERROR[03-11|00:17:49.080|core/vm/evm.go:583] Error in calling the EVM                 funcName=getAddressFor transac
tionData=0xdd9272338b91d837e1684f7353d73b6197230894243cf869282f722841df96b441303f37 err="evm: execution reverted"
ERROR[03-11|00:17:49.080|core/vm/evm.go:583] Error in calling the EVM                 funcName=getAddressFor transac
tionData=0xdd927233d7e89ade8430819f08bf97a087285824af3351ee12d72a2d132b0c6c0687bfaf err="evm: execution reverted"
ERROR[03-11|00:17:49.080|core/vm/evm.go:583] Error in calling the EVM                 funcName=getAddressFor transac
tionData=0xdd92723348766a30bcca8b3c8b03b4e4b1f6510aee73f7b41aa6c7578a97040d20334f42 err="evm: execution reverted"
ERROR[03-11|00:17:49.195|core/vm/evm.go:583] Error in calling the EVM                 funcName=getAddressFor transac
tionData=0xdd927233235a6f54090e9b94aa4e585a699c4375a2ff8f572c68114d138f0ed121527849 err="evm: execution reverted"
ERROR[03-11|00:18:01.928|core/vm/evm.go:583] Error in calling the EVM                 funcName=frozen        transac
tionData=0x054f7d9c err="evm: execution reverted"
ERROR[03-11|00:18:01.928|contract_comm/evm.go:124] Error when invoking evm function         err="evm: execution reve
rted" funcName=frozen        static=true address=0x74B0b1139Bf7DF2Aea998Af47c41a683276f28C1 args=[] gas=20000 gasLef
t=17850 value=<nil>
ERROR[03-11|00:18:01.928|contract_comm/evm.go:159] Error in executing function on registered contract function=froze
n registryId=0xb33c3d77234979a288baf651b42d19d8483d5af3e328f9f1fae2bef7b11acd25 err="evm: execution reverted"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x486e173]

goroutine 323 [running]:
github.com/ethereum/go-ethereum/consensus/istanbul/backend.(*Backend).updateValidatorScores(0xc0003b1340, 0xc005ba34
40, 0xc0013599e0, 0xc0053d1580, 0x4, 0x4, 0xc0053d1580, 0x4, 0x4, 0x0, ...)
        /Users/justin/celo-blockchain/build/_workspace/src/github.com/ethereum/go-ethereum/consensus/istanbul/backend/pos.go:139 +0x373
github.com/ethereum/go-ethereum/consensus/istanbul/backend.(*Backend).distributeEpochRewards(0xc0003b1340, 0xc005ba3440, 0xc0013599e0, 0x0, 0x0)
        /Users/justin/celo-blockchain/build/_workspace/src/github.com/ethereum/go-ethereum/consensus/istanbul/backend/pos.go:75 +0x60f
github.com/ethereum/go-ethereum/consensus/istanbul/backend.(*Backend).Finalize(0xc0003b1340, 0x52afe00, 0xc000374000, 0xc005ba3440, 0xc0013599e0, 0xc0054de650, 0x1, 0x1, 0x5bfb550, 0x0, ...)
        /Users/justin/celo-blockchain/build/_workspace/src/github.com/ethereum/go-ethereum/consensus/istanbul/backend/engine.go:472 +0x435
github.com/ethereum/go-ethereum/core.(*StateProcessor).Process(0xc00000ebe0, 0xc005a277c0, 0xc0013599e0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /Users/justin/celo-blockchain/build/_workspace/src/github.com/ethereum/go-ethereum/core/state_processor.go:90 +0x76b
github.com/ethereum/go-ethereum/core.(*BlockChain).insertChain(0xc000374000, 0xc00519dc20, 0x11, 0x11, 0x2a1168b14ee9d201, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /Users/justin/celo-blockchain/build/_workspace/src/github.com/ethereum/go-ethereum/core/blockchain.go:1738 +0xe2d
github.com/ethereum/go-ethereum/core.(*BlockChain).InsertChain(0xc000374000, 0xc00519dc20, 0x11, 0x11, 0x0, 0x0, 0x0)
        /Users/justin/celo-blockchain/build/_workspace/src/github.com/ethereum/go-ethereum/core/blockchain.go:1552 +0xd01
github.com/ethereum/go-ethereum/eth/downloader.(*Downloader).importBlockResults(0xc0003e6000, 0xc003911300, 0x11, 0x20, 0xb4a891695506c000, 0xc004b9c850)
        /Users/justin/celo-blockchain/build/_workspace/src/github.com/ethereum/go-ethereum/eth/downloader/downloader.go:1691 +0x4c9
github.com/ethereum/go-ethereum/eth/downloader.(*Downloader).processFastSyncContent(0xc0003e6000, 0xc0013cf440, 0x0, 0x0)
        /Users/justin/celo-blockchain/build/_workspace/src/github.com/ethereum/go-ethereum/eth/downloader/downloader.go:1793 +0x51d
github.com/ethereum/go-ethereum/eth/downloader.(*Downloader).syncWithPeer.func7(0xc0012a7f70, 0x0)
        /Users/justin/celo-blockchain/build/_workspace/src/github.com/ethereum/go-ethereum/eth/downloader/downloader.go:550 +0x33
github.com/ethereum/go-ethereum/eth/downloader.(*Downloader).spawnSync.func1(0xc0003e6000, 0xc00149c300, 0xc0014a0f40)
        /Users/justin/celo-blockchain/build/_workspace/src/github.com/ethereum/go-ethereum/eth/downloader/downloader.go:564 +0x63
created by github.com/ethereum/go-ethereum/eth/downloader.(*Downloader).spawnSync
        /Users/justin/celo-blockchain/build/_workspace/src/github.com/ethereum/go-ethereum/eth/downloader/downloader.go:564 +0xaf
```

**Pull request merged**